pkg/tun/tcp/handler.go

```diff
@@ -139,6 +139,11 @@ type handler struct {
 	sendLock      sync.Mutex
 	sendCondition *sync.Cond
 
+	// isClosing indicates whether Close() has been called on the handler
+	isClosing bool
```

Comment:
This variable is written and read from different goroutines so it needs a mutex or be atomic. Perhaps you can use an `int32` instead of a `bool` and then `atomic.StoreInt32()` and `atomic.LoadInt32()` to store and test.